crypto/types/types.go

```diff
@@ -22,3 +22,9 @@ type PrivKey interface {
 type (
 	Address = tmcrypto.Address
 )
+
+// IntoTmPubKey allows our own PubKey types be converted into Tendermint's
+// pubkey types.
+type IntoTmPubKey interface {
```

Comment:
Should this also house the `From` method?

Reply:
I added this function [`FromTmEd25519(pubKey crypto.PubKey) (*PubKey, error)`](https://github.com/cosmos/cosmos-sdk/pull/7268/files#diff-26daf622d7b5fc55c7b92e2e4816692aR225)

Not sure about putting it in the interface:
```golang
// right now
valPubKey, err := ed25519.FromTmEd25519(tmValPubKey)

// by putting it in the interface
var valPubKey ed25519.PubKey
valPubKey = valPubKey.FromTmPubKey(tmValPubKey) // reads funny
```